components/CounterButton/index.js

```diff
@@ -0,0 +1,55 @@
+import { StyledSpan } from "../../styles/StyledSpan";
+import { StyledButton } from "../../styles/StyledButton";
+import { StyledInstruction } from "../../styles/StyledInstruction";
+import { StyledInstruction2 } from "../../styles/StyledInstruction2";
+import useStore from "../useStore";
+import useLoading from "../useLoading";
+
+export default function CounterButton({
+  handleBreathClick,
+  handleRetentionCounterClick,
+  displayTime,
+}) {
+  const breathCount = useStore((state) => state.breathCount);
+  const isBreathActive = useStore((state) => state.isBreathActive);
+  const showRetentionCounter = useStore((state) => state.showRetentionCounter);
+  const isRetentionFinished = useStore((state) => state.isRetentionFinished);
+  const breathHoldCountdown = useStore((state) => state.breathHoldCountdown);
+  const isLoading = useStore((state) => state.isLoading);
+  const switchIsLoading = useStore((state) => state.switchIsLoading);
+
+  useLoading(switchIsLoading);
+
+  return (
+    <>
+      {showRetentionCounter ? (
+        <StyledButton onClick={handleRetentionCounterClick} retention>
+          <StyledInstruction>Hold breath</StyledInstruction> {displayTime}
+        </StyledButton>
+      ) : isRetentionFinished ? (
+        <StyledButton isRetentionFinished>
```

Comment:
There is no `onClick` prop here, what does this button do?

Reply:
The serves as main element focusing on all main functionality. Therefore the button is clickable at the beginning and during a certain part of the flow to save the breath retention count. During the exercise itself it is disabled and showing the instructions ("hold breath" etc.) as well as displaying the counter. Therefore it is on purpose, that at some points of the flow the button is to click while during the counting itself it just displays the count and infos :) The instruction to the exercise is also explained on the info page.